### PR TITLE
add: returnRawResult option

### PR DESCRIPTION
**Pull request by june07**
Currently only the SuccessValue is returned from calling contracts with this api...

https://github.com/near/near-api-js/blob/a05667361ddab208423f5a5c0e3dd0ce182b9880/lib/providers/provider.js#L30

This option provides a means to access the full rawResult object.

### CODE REVIEW

**Comment by june07** [comment]
Tests fine now in my environment.

File: src/contract.ts [hunk 1]
```diff
@@ -103,36 +103,39 @@ export class Contract {
...
-                            methodName,
-                            args: args[0],
-                            gas: args[1],
-                            amount: args[2]
-                        });
-                    }
+                        if(args.length > 1 || !(args[0] && args[0].args)) {
+                            const deprecate = depd('contract.methodName(args, gas, amount)');
+                            deprecate('use `contract.methodName({ args, gas?, amount?, callbackUrl?, meta? })` instead');
+                            return this[resultType === '' ? '_changeMethod' : '_changeMethodRaw']({
```
Comment (willemneal) [inline src/contract.ts:120]: Small nit that this line is repeated below and could be refactored above to select the property.

File: src/contract.ts [hunk 2]
```diff
@@ -142,8 +145,15 @@ export class Contract {
             walletCallbackUrl: callbackUrl
         });
 
-        return getTransactionLastResult(rawResult);
+        return result;
+    }
+    
+    private async _changeMethod({ args, methodName, gas, amount, meta, callbackUrl }: ChangeMethodOptions) {
+        const result = await this._changeMethodRaw({ args, methodName, gas, amount, meta, callbackUrl });
+
```
Comment (willemneal) [inline src/contract.ts:153]:
```suggestion
```

**Review by willemneal — APPROVED** [review]
I approve this!

**Comment by june07** [comment]
> I approve this!

Just remove my yalc version, so it'll be great when this is pushed.

**Comment by willemneal** [comment]
@june07 While I approve, that's all I can do.

**Comment by june07** [comment]
@willemneal Much appreciated!

**Comment by june07** [comment]
@vgrichina How does it look?

**Comment by june07** [comment]
Bump @vgrichina

**Comment by june07** [comment]
bump again

**Comment by june07** [comment]
bump... again...

**Comment by willemneal** [comment]
Ping

**Comment by june07** [comment]
Pong  :)

**Comment by june07** [comment]
Jesus Christ!  Is there a problem with the PR or what?!  Two months is beyond rediculous.

**Comment by june07** [comment]
Any feedback on this one? @willemneal maybe you have some insight on why this was never merged?

**Comment by willemneal** [comment]
@june07 It seems that they are waiting to hire someone to maintain this project.  Thank you for your continued efforts in improving NAJ.  My guess is that since they have plans for a big rewrite (to break this project into smaller pieces) and with their limited resources (mostly going toward Developer Console and workspaces-js), maintenance work is a low priority.

@ilblackdragon @nearmax @MaximusHaximus Any news on the hiring front? We wouldn't want to lose future OS contributors.

**Comment by adhbh** [comment]
Let's push this PR! We require this feature as well

**Pull request closed without merging**